platform-hub-api/app/controllers/kubernetes/clusters_controller.rb

```diff
@@ -1,52 +1,42 @@
 class Kubernetes::ClustersController < ApiJsonController
 
-  before_action :load_kubernetes_clusters_hash_record
-
   # GET /kubernetes/clusters
   def index
     authorize! :read, :kubernetes_clusters
-    clusters = @kubernetes_clusters.data.map do |c|
-      c.with_indifferent_access.slice(:id, :description)
-    end.sort_by! do |c|
-      c[:id]
-    end
-    render json: clusters
+    clusters = Kubernetes::ClusterService.all_clusters.sort_by(&:name)
+    render json: clusters, each_serializer: KubernetesClusterSerializer
   end
 
-  # PATCH/PUT /kubernetes/clusters/:id
+  # PATCH/PUT /kubernetes/clusters/:name
   def create_or_update
     authorize! :manage, :kubernetes_clusters
 
     data = cluster_params.to_h
-    data[:id] = params[:id]  # ID in URL takes precedence
+    data[:name] = params[:name]  # Cluster name in URL takes precedence
 
     Kubernetes::ClusterService.create_or_update data
 
     AuditService.log(
       context: audit_context,
       action: 'update_kubernetes_cluster',
```

Comment:
We should consider changing this action now to be the regular CRUD actions that we do for all other AR based resources.

Also, need to set the `cluster` object in the Audit now as the `auditable` object.